### PR TITLE
feat(component): generate a simple component by default

### PR DESCRIPTION
**Pull request by matthieu-crouzet**
## Proposed change

Generate a simple component by default

### CODE REVIEW

**Review by pginoux-1A — DISMISSED** [review]

File: packages/@o3r/core/schematics/component/presenter/index.ts [hunk 1]
```diff
@@ -66,10 +67,14 @@ export function ngGenerateComponentPresenter(options: NgGenerateComponentSchemat
...
   const generateFiles = (tree: Tree, _context: SchematicContext) => {
-
+    _context.logger.debug(JSON.stringify(options));
     const workspaceProject = options.projectName ? getWorkspaceConfig(tree)?.projects[options.projectName] : undefined;
 
-    const properties = getTemplateProperties(options, ComponentStructureDef.Pres, options.prefix ? options.prefix : workspaceProject?.prefix);
+    const properties = getTemplateProperties(
+      options,
+      options.componentStructure === 'simple' ? ComponentStructureDef.Simple : ComponentStructureDef.Pres,
+      options.prefix ? options.prefix : workspaceProject?.prefix
```
Comment (fpaul-1A) [inline packages/@o3r/core/schematics/component/presenter/index.ts:76]:
```suggestion
      options.prefix || workspaceProject?.prefix
```

File: packages/@o3r/core/schematics/component/presenter/index.ts [hunk 2]
```diff
@@ -66,10 +67,14 @@ export function ngGenerateComponentPresenter(options: NgGenerateComponentSchemat
   const fullStructureRequested = options.componentStructure === 'full';
 
   const generateFiles = (tree: Tree, _context: SchematicContext) => {
-
+    _context.logger.debug(JSON.stringify(options));
```
Comment (vscaiceanu-1a) [inline packages/@o3r/core/schematics/component/presenter/index.ts:70]: do you need this?

File: packages/@o3r/core/schematics/component/presenter/index.ts [hunk 3]
```diff
@@ -44,15 +44,16 @@ export const PRESENTER_FOLDER = 'presenter';
...
   return {
     ...options,
-    componentName: getComponentName(inputComponentName, componentStructureDef).replace(/Component$/, ''),
+    componentName: getComponentName(inputComponentName, structure).replace(/Component$/, ''),
     componentSelector: getComponentSelectorWithoutSuffix(options.componentName, prefix || null),
     projectName: options.projectName || getLibraryNameFromPath(options.path),
     folderName,
-    name: getComponentFileName(options.componentName, componentStructureDef), // air-offer | air-offer-pres
-    suffix: componentStructureDef.toLowerCase(), // pres | ''
+    name: getComponentFileName(options.componentName, structure), // air-offer | air-offer-pres
```
Comment (vscaiceanu-1a) [inline packages/@o3r/core/schematics/component/presenter/index.ts:55]:
```suggestion
    name: getComponentFileName(options.componentName, structure),
```

File: packages/@o3r/core/schematics/component/index.ts [hunk 4]
```diff
@@ -93,16 +93,24 @@ export function ngGenerateComponent(options: NgGenerateComponentSchematicsSchema
...
         generateRootBarrel,
         generateComponentPresenter(options)
       ]);
+    }
+    case 'presenter': {
+      return chain([
+        generateRootBarrel,
+        generateComponentPresenter(options)
+      ]);
+    }
```
Comment (fpaul-1A) [inline packages/@o3r/core/schematics/component/index.ts:113]:
```suggestion
    case 'simple':
    case 'presenter': {
      return chain([
        generateRootBarrel,
        generateComponentPresenter(options)
      ]);
    }
```

File: packages/@o3r/core/schematics/component/presenter/index.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (matthieu-crouzet) [reply packages/@o3r/core/schematics/component/presenter/index.ts:70]: nope :)

**Review by vscaiceanu-1a — DISMISSED** [review]

**Review by fpaul-1A — APPROVED** [review]

**Review by kpanot — DISMISSED** [review]

**Review by fpaul-1A — APPROVED** [review]

**Review by kpanot — APPROVED** [review]

**Pull request merged**